app/src/main/java/org/oppia/app/onboarding/OnboardingFragmentPresenter.kt

```diff
@@ -0,0 +1,114 @@
+import androidx.viewpager.widget.ViewPager
+import kotlinx.android.synthetic.main.onboarding_fragment.view.*
+import org.oppia.app.R
+import org.oppia.app.databinding.OnboardingFragmentBinding
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.viewmodel.ViewModelProvider
+import javax.inject.Inject
+
+/** The presenter for [OnboardingFragment]. */
+@FragmentScope
+class OnboardingFragmentPresenter @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val fragment: Fragment,
+  private val viewModelProvider: ViewModelProvider<OnboardingViewModel>
+) {
+  private val dotsList = ArrayList<ImageView>()
+  private lateinit var onboardingPagerAdapter: OnboardingPagerAdapter
+  private val routeToProfileListener = activity as RouteToProfileListener
+  private lateinit var binding: OnboardingFragmentBinding
+  private lateinit var slidesViewPager: ViewPager
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
+    binding = OnboardingFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
+    // NB: Both the view model and lifecycle owner must be set in order to correctly bind LiveData elements to
+    // data-bound view models.
+    binding.let {
+      it.lifecycleOwner = fragment
+      it.presenter = this
+      it.viewModel = getOnboardingViewModel()
+    }
+    slidesViewPager = binding.root.onboarding_slide_view_pager as ViewPager
+    setUpViewPager(slidesViewPager)
+    addDots()
+    return binding.root
+  }
+
+  private fun setUpViewPager(viewPager: ViewPager) {
+    onboardingPagerAdapter = OnboardingPagerAdapter(fragment.requireContext())
+    viewPager.adapter = onboardingPagerAdapter
+    viewPager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
+      override fun onPageScrollStateChanged(state: Int) {
+      }
+
+      override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
+      }
+
+      override fun onPageSelected(position: Int) {
+        getOnboardingViewModel().slideChanged(position)
+        selectDot(position)
+      }
+    })
+  }
+
+  fun clickOnGetStarted() {
+    routeToProfileListener.routeToProfile()
+  }
+
+  fun clickOnSkip() {
+    getOnboardingViewModel().slideChanged(3)
+    slidesViewPager.currentItem = 3
+  }
+
+  private fun getOnboardingViewModel(): OnboardingViewModel {
+    return viewModelProvider.getForFragment(fragment, OnboardingViewModel::class.java)
+  }
+
+  private fun addDots() {
+    val dotsLayout = binding.slideDotsContainer
+    val dotIdList = ArrayList<Int>()
+    dotIdList.add(R.id.onboarding_dot_0)
+    dotIdList.add(R.id.onboarding_dot_1)
+    dotIdList.add(R.id.onboarding_dot_2)
+    dotIdList.add(R.id.onboarding_dot_3)
+    for (index in 0 until TOTAL_NUMBER_OF_SLIDES) {
+      val dotView = ImageView(activity)
+      dotView.id = dotIdList[index]
+      dotView.setImageResource(R.drawable.onboarding_dot_active)
+      if (index != 0) {
```

Comment:
This seems a bit obscure to me. Perhaps it would be cleaner to omit this and call selectDot at the end of this method to select the initial page?

Reply:
Yeah, that makes sense actually. Removed this if-else and after for loop, called `selectDot(0)`